scalableminds/webknossos
# Readd zarr3 streaming v0

Revision of Zarr 3 streaming.
The error was caused due to expecting a prefixed "c." when addressing an array. I now changed the chunk key encoding of zarr 3 to "v2" to omit this prepended "c." just as it was done in zarr 2 streaming. Now the coordinate parsing code is slightly easier and works with v2 and v3.


Please just review the following commits. They are the only changes compared to the first version of this pr:
- [e217343](https://github.com/scalableminds/webknossos/pull/7995/commits/e217343e2107e20674f047d94db5bd046ecf678d)
- [7438f2c](https://github.com/scalableminds/webknossos/pull/7995/commits/7438f2c14cef15539eb5570db3336218b6c8253c)
- [c4ecf5a](https://github.com/scalableminds/webknossos/pull/7995/commits/c4ecf5adec790cb8d2bdeecd3724f969fb1a3e77)
- [9e1bed0](https://github.com/scalableminds/webknossos/pull/7995/commits/9e1bed09ddfd615ea236dce4aa4f74ff87bf63c8)


### URL of deployed dev instance (used for testing):
- https://addzarrstreamingv.webknossos.xyz/

### Steps to test:
- Test the following streamed datasets:
  - Zarr 2 view mode streamed: https://addzarrstreamingv.webknossos.xyz/datasets/sample_organization/l4_sample_view_streamed_zarrv2/view#3584,3584,1024,0,1.3
  - Zarr 3 view mode streamed: https://addzarrstreamingv.webknossos.xyz/datasets/sample_organization/l4_sample_view_streamed_zarrv3/view#3584,3584,1024,0,1.3
  -  Zarr 2 annotation streamed (should have visible changes in the volume layer): https://addzarrstreamingv.webknossos.xyz/datasets/sample_organization/l4_sample_annotation_streamed_zarrv2/view#3584,3584,1024,0,1.3
  - Zarr 3 annotation streamed: https://addzarrstreamingv.webknossos.xyz/datasets/sample_organization/l4_sample_annotation_streamed_zarrv3/view#3584,3584,1024,0,1.3

### TODOs:
- [x] Deploy dev instance and create zarr2 & zarr3 streaming test datasets for view only and annotations

### Issues:
- fixes #7933

------
(Please delete unneeded items, merge only when none are left open)
